src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable default-case */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useRef } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  toDelete: (id: number) => void;
+  toToggle: (id: number) => void;
+  toChangeTitle: (title: string, id: number) => void;
+};
+
+enum EventTypes {
+  Blur = 'blur',
+  Escape = 'escape',
+  Enter = 'enter',
+}
+
+export const TodoItem: React.FC<Props> = React.memo((({
+  todo, toDelete, toToggle, toChangeTitle,
+}) => {
+  const { id, title, completed } = todo;
+  const timer = useRef<ReturnType<typeof setInterval> | null>(null);
+
+  const applyTodoTitleChanges = (
+    input: HTMLInputElement,
+    event: EventTypes,
+  ) => {
+    const editedTodo = input.closest('li');
+
+    switch (event) {
+      case EventTypes.Escape:
+      case EventTypes.Enter:
+      case EventTypes.Blur:
+        toChangeTitle(input.value, id);
+        break;
+      default:
+        toChangeTitle('', id);
+    }
+
+    if (editedTodo) {
+      editedTodo.classList.remove('editing');
+    }
+  };
+
+  // Handler for identifying single click and double click. And opened input for editing todo
+  const handlerClick = (event: React.MouseEvent) => {
+    if (timer.current) {
```

Comment:
Avoid using timers in event handlers when possible. This can lead to confusing behavior and performance issues.